docs/0002-choose-a-pagination-strategy-and-implementation.md

```diff
@@ -0,0 +1,101 @@
+[existing](https://developer.github.com/v3/#pagination) [resources](https://developer.spotify.com/documentation/web-api/reference/object-model/#paging-object).
+
+### limit & offset
+
+Using `offset` and `limit` parameters is a common patten for paginating results. With this approach the `limit` 
+parameter specifies the number of results in a response and the `offset` parameter specifies which particular set of 
+results out of the total to return. This maps closely to how `limit` and `offset` work in `SQL` already.
+
+#### Benefits
+
+**Easy to implement**: maps closely to `SQL` and therefore easy to implement
+
+**Stateless**: requires no additional context
+
+#### Disadvantages
+
+**Performance**: can be slow for large datasets
+
+**Inconsistent**: depending on sort ordering and the inclusion/deletion of items the same results can be returned multiple times
+
+### keyset/cursor/continuation token
+
+The `keyset`, `cursor`, or `continuation` token approach uses an opaque-*ish* token to control the page of results 
+returned and can also be combined with a `limit` parameter to control the number of results. Often the token used will 
+represent a unique identifier for the last result of the current page returned to the client. This token is then used in
+subsequent requests use the token in a query parameter (e.g. `/?next=<token>`).
+
+#### Benefits
+
+**Performance**: on large datasets with indices performs better than scanning previous results
+
+**Consistent Ordering**: even when data is added/removed results are stable
+
+**Can page forward and backwards**: tokens can be used to page before and after search results
+
+**Implementation logic hidden from users**: while there may be an interpretation of a continuation token, it is not
+guaranteed to be consistent or long-lived, which frees up back-end optimizations and changes without breaking the API
+
+#### Disadvantages
+**Implementation more complex**: needs to manage sort ordering to ensure that the continuation token equality check is
+consistent even if data is added, removed, or sorted by additional fields
+
+**Tokens become invalid**: continuation tokens can become invalid, users should not rely on them for long periods of 
+time
+
+## Decision
+
+We will use **continuation** tokens for managing pagination. The advantage of keeping the implementation flexible/hidden
+from end-users so that it can be optimized or changed later without breaking functionality for existing users outweighs
+disadvantages around increased complexity for implementation. Additionally, since `franlin` APIs are expected to hold 
+large amounts of data (tens of millions of rows) it is important that the pagination approach scales accordingly.
+Lastly, since eventually `franklin` may move beyond relying solely on `postgresl` for managing state it may actually be
```

Comment:
This is a really good point and I think valuable to have in the open on the docs page for bringing other people to the party at future STAC sprints. I'm curious about whether we'll learn enough in Practical FP in Scala to end up with a `CollectionAlgebra`, `ItemAlgebra`, etc. that we can use to support different backends better

Reply:
That's my hope for the Practical FP In Scala book too -- that once we've implemented a final tagless encoding approach in a toy application with training wheels applying it to franklin won't be too scary.